### PR TITLE
Adds pets as traitor objectives for the syndicate to hold ransom.

### PR DESCRIPTION
**Pull request by SkippuSetsuna**
## About The Pull Request
~~Brings back the actual budget cards back into the head of staff's lockers and is now a target of the syndicate. Stamps are also an objective to steal as well as the HoP's pet, Ian.~~

Adds some head of staff pets to traitor objectives

## Why It's Good For The Game
Adds more objectives to the game for fun gameplay experinces. 

## Changelog
:cl:
add: some pets to the traitor objective list
/:cl:

### CODE REVIEW

**Comment by SkippuSetsuna** [comment]
Probably going to add other head's pets to the list later.

**Comment by ATH1909** [comment]
Pretty much the only thing I like in this PR is the addition of Ian as a potential theft objective.

The non-cargo budget cards were removed from the head of staff lockers for a REASON. With them in the game, Cargo doesn't have to do any bounties or sell anything- the amount of money gained from an entire round's worth of bounties is approximately equivalent to the amount of money that Cargo can get by asking ONE head of staff (the RD) to hand over a budget card that that head of staff mostly doesn't need. Cargo can do this near ROUNDSTART, as well; being able to go full strength Gun Cargo 5 minutes into the round is not exactly what I'd call healthy for game/Cargo balance. They also allow heads of staff to basically ignore the economy system, so the CMO can buy 4 advanced medkits and insulated gloves with their shiftstart money, for example.

_Stamps_ should not be traitor theft objective items. Nobody cares about stamps (except for validhunters or Sec when they search you), and no one will notice their absence if someone takes them... Unless you're trying to steal the QM's stamp, which I could _maybe_ see as being an okay theft objective, although I feel like it's a bit TOO important for Cargo for it to be a commonly stolen item. Traitor theft objectives should be strong items that have legitimate non-traitor objective uses/benefits (the CMO's hypospray, the RD's reactive tele armor, the HoS's X-01 laser, etc.) that won't significantly impair an entire department if stolen (although it'll surely piss off the head of that department). What, exactly, are you supposed to use a _stamp_ for, other than for stamping Cargo manifests or for saying things like "I got approval from heads X, Y, and Z, please give me AA"?

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [inline code/game/gamemodes/objective_items.dm]: widget

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [inline code/game/gamemodes/objective_items.dm:142]: these should have a common parent

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [inline code/game/gamemodes/objective_items.dm]: araneus is only a thing on metastation and you use any bat to get the objective

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [inline code/game/gamemodes/objective_items.dm]: nothing personnel kid

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [inline code/game/gamemodes/objective_items.dm]: this doesnt even work

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [inline code/game/gamemodes/objective_items.dm]: thats not the job name

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [inline code/game/gamemodes/objective_items.dm]: not here either!

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [inline code/game/gamemodes/objective_items.dm]: ah yes, science, my favorite engineering job

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [reply code/game/gamemodes/objective_items.dm]: also miners unlike cargo techs actually DO get qm access on deadpop so i dont see why you didnt put them there
actually why do you even put people that aren't the heads there

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [inline code/game/gamemodes/objective_items.dm]: not capitalized

File: code/game/gamemodes/objective_items.dm
Comment (SkippuSetsuna) [reply code/game/gamemodes/objective_items.dm]: Oof didn't notice that.

File: code/game/gamemodes/objective_items.dm
Comment (SkippuSetsuna) [reply code/game/gamemodes/objective_items.dm]: It was pretty ify to add this, but I'll probablly remove it.

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [reply code/game/gamemodes/objective_items.dm:142]: also would be cool if you added a lamar objective

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [inline code/game/gamemodes/objective_items.dm]: you forgot geneticist here

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [inline code/game/gamemodes/objective_items.dm]: and sec/detective here

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [inline code/game/gamemodes/objective_items.dm]: this shouldnt exclude jobs

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [inline code/game/gamemodes/objective_items.dm]: its the head of personnel

File: code/game/gamemodes/objective_items.dm
Comment (SkippuSetsuna) [reply code/game/gamemodes/objective_items.dm:142]: Yeah! I'll probably will. Thanks for the suggestion.
I do have a question however as I'm pretty new. How do I set the name for the objective with each corresponding animal?

**Comment by SkippuSetsuna** [comment]
> 
> 
> Pretty much the only thing I like in this PR is the addition of Ian as a potential theft objective.
> 
> The non-cargo budget cards were removed from the head of staff lockers for a REASON. With them in the game, Cargo doesn't have to do any bounties or sell anything- the amount of money gained from an entire round's worth of bounties is approximately equivalent to the amount of money that Cargo can get by asking ONE head of staff (the RD) to hand over a budget card that that head of staff mostly doesn't need. Cargo can do this near ROUNDSTART, as well; being able to go full strength Gun Cargo 5 minutes into the round is not exactly what I'd call healthy for game/Cargo balance. They also allow heads of staff to basically ignore the economy system, so the CMO can buy 4 advanced medkits and insulated gloves with their shiftstart money, for example.
> 
> _Stamps_ should not be traitor theft objective items. Nobody cares about stamps (except for validhunters or Sec when they search you), and no one will notice their absence if someone takes them... Unless you're trying to steal the QM's stamp, which I could _maybe_ see as being an okay theft objective, although I feel like it's a bit TOO important for Cargo for it to be a commonly stolen item. Traitor theft objectives should be strong items that have legitimate non-traitor objective uses/benefits (the CMO's hypospray, the RD's reactive tele armor, the HoS's X-01 laser, etc.) that won't significantly impair an entire department if stolen (although it'll surely piss off the head of that department). What, exactly, are you supposed to use a _stamp_ for, other than for stamping Cargo manifests or for saying things like "I got approval from heads X, Y, and Z, please give me AA"?

After sleeping on this last night I realized that I wanted to shift this to more objective-based than actually reverting the budget code. Pretty ecstatic about including pets as targets.

File: code/game/gamemodes/objective_items.dm
Comment (SkippuSetsuna) [reply code/game/gamemodes/objective_items.dm]: The departmental jobs, not the head right?

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [inline code/game/gamemodes/objective_items.dm]: Quartermaster

File: code/game/gamemodes/objective_items.dm
Comment (Fikou) [reply code/game/gamemodes/objective_items.dm:142]: objective names can stay different but you can do something like
"[pet.name], alive."
with a pet var on the parent of course

File: code/game/gamemodes/objective_items.dm
Comment (Shadowtail117) [inline code/game/gamemodes/objective_items.dm]: Not a fan of this change. If the cargo budget card gets compromised, the entire department becomes virtually useless. The traitor (or the Syndicate) can siphon all funds from the budget and now no one can buy anything unless they pay out-of-pocket, which is way too expensive even for most heads of staff.

**Review by Shadowtail117 — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/objective_items.dm
Comment (ATH1909) [reply code/game/gamemodes/objective_items.dm]: "buget"

File: code/game/gamemodes/objective_items.dm
Comment (remove32) [reply code/game/gamemodes/objective_items.dm]: why would that be unacceptable?

**Comment by remove32** [comment]
thanks for reverting the bit about the budget cards. that wouldn't have been good at all. i know it was a controversial removal for some people but it was necessary

File: code/game/gamemodes/objective_items.dm
Comment (ATH1909) [reply code/game/gamemodes/objective_items.dm]: Because it encourages traitors to steal something that can fuck over an entire department.

As a cargonian, what are you supposed to do if someone steals the budget card and siphons money out of it every time you sell something? You have no way of tracking down where the card is or identifying who the thief is (if they've been smart and haven't left fingerprints everywhere), so you're just kinda screwed.

Comment (Shadowtail117) [reply code/game/gamemodes/objective_items.dm]: Unlike other departmental high-risk items, like the HoS' laser gun or the CMO's hypospray, the cargo budget is essential to the entire department, and without it it cannot function. No other objective (besides hijack, I suppose) does that. Given other departments could still function even when their areas are destroyed (medical can still heal and do surgery outside of medical, security is usually outside of security anyway, engineering can always move somewhere else like auxilary power, etc.), giving traitors the objective to completely disable an entire department seems completely unbalanced.

Comment (SkippuSetsuna) [reply code/game/gamemodes/objective_items.dm]: I’ll remove it whenever I get the chance to later tonight after work.

**Review by Shadowtail117 — APPROVED** [review]

**Comment by TheSwain** [comment]
HEY GUYS now that I'm back to reviewing PRs, do we have a verdict on this one? is there anything pending before we merge? did we like this?

File: code/game/gamemodes/objective_items.dm
Comment (remove32) [inline code/game/gamemodes/objective_items.dm:152]: you dont seem to be checking that the objective is anywhere - this proc should return true so long as Ian is alive in a pet carrier somewhere

**Review by remove32 — DISMISSED** [review]

File: code/game/gamemodes/objective_items.dm
Comment (remove32) [reply code/game/gamemodes/objective_items.dm:152]: turns out i am stupid and this works just fine. check_special_completion must have some other code i'm not seeing to check if the object is on you

**Review by remove32 — APPROVED** [review]

**Pull request closed without merging**
